tests/sensors/test_weekday_sensor.py

```diff
@@ -33,130 +33,91 @@
 WEEKEND_DATE = datetime(2018, 12, 22)
 TEST_DAG_ID = 'weekday_sensor_dag'
 DEV_NULL = '/dev/null'
-
-
-class TestDayOfWeekSensor(unittest.TestCase):
-
-    def setUp(self):
-        self.dagbag = DagBag(
-            dag_folder=DEV_NULL,
-            include_examples=True
-        )
-        self.args = {
-            'owner': 'airflow',
-            'start_date': DEFAULT_DATE
-        }
-        dag = DAG(TEST_DAG_ID, default_args=self.args)
-        self.dag = dag
-
-    def tearDown(self):
-        session = Session()
-        session.query(models.TaskInstance).filter_by(
-            dag_id=TEST_DAG_ID).delete()
-        session.query(TaskFail).filter_by(
-            dag_id=TEST_DAG_ID).delete()
-        session.commit()
-        session.close()
-
-    def test_weekday_sensor_true(self):
-        op = DayOfWeekSensor(
-            task_id='weekday_sensor_check_true',
-            week_day='Thursday',
-            use_task_execution_day=True,
-            dag=self.dag)
+TRUE_TESTS_CONFIG = {
+    "string": "Thursday",
+    "enum": WeekDay.THURSDAY,
+    "enum-set": {WeekDay.THURSDAY},
+    "enum-set-2-items": {WeekDay.THURSDAY, WeekDay.FRIDAY},
+    "string-set": {'Thursday'},
+    "string-set-2-items": {'Thursday', 'Friday'},
+}
+
+
+@pytest.fixture
+def dag():
+    args = {"owner": "airflow", "start_date": DEFAULT_DATE}
+    return DAG(TEST_DAG_ID, default_args=args)
+
+
+@pytest.fixture
+def tear_down():
+    yield
+
+    session = Session()
+    session.query(models.TaskInstance).filter_by(
+        dag_id=TEST_DAG_ID).delete()
+    session.query(TaskFail).filter_by(
+        dag_id=TEST_DAG_ID).delete()
+    session.commit()
+    session.close()
+
+
+@pytest.mark.parametrize("weekday", TRUE_TESTS_CONFIG.values(), ids=list(TRUE_TESTS_CONFIG.keys()))
+def test_weekday_sensor_true(dag, weekday, tear_down):
+    op = DayOfWeekSensor(
+        task_id="weekday_sensor_check_true",
```

Comment:
It is necessary to use classes in tests. More information: https://lists.apache.org/thread.html/1e4df7d4b0cd9b2d2bb76a3336471aa85e852545dd41ada6d4e461b8%40%3Cdev.airflow.apache.org%3E

Reply:
Ah, that makes sense. Thank you for the heads up 👍

Reply:
This helped me learn about how tests **are** parameterized here in `airflow`. Went with a different approach over in #7316.